src/minmax.rs

```diff
@@ -0,0 +1,90 @@
+use rustc::lint::{Context, LintPass, LintArray};
+use rustc_front::hir::*;
+use syntax::codemap::Spanned;
+use syntax::ptr::P;
+use std::cmp::PartialOrd;
+use std::cmp::Ordering::*;
+
+use consts::{Constant, constant};
+use utils::{match_path, span_lint};
+use self::MinMax::{Min, Max};
+
+declare_lint!(pub MIN_MAX, Deny,
```

Comment:
I have removed the resolution. Also match on full path.

Again, the result of the linted expression is *constant*. Unless someone is being sneaky, the code is certainly not working as intended. Most possibly whoever writes such code has either switched the constants or the `min`/`max` erroneously. I see no useful case for such code (except perhaps an Underhanded Rust Contest). Do you?